src/components/search/data/hooks.js

```diff
@@ -143,7 +144,7 @@ export const useDefaultSearchFilters = ({ enterpriseConfig, subscriptionPlan, of
       // if there's a subscriptionPlan, filter results by the subscription catalog
       // and any catalogs for which the user has vouchers
       if (subscriptionPlan) {
-        if (offerCatalogs.length > 0) {
+        if (features.ENROLL_WITH_CODES && offerCatalogs.length > 0) {
```

Comment:
On line 153, I think we'll want to feature flag the default case to not consider offers and instead default to show all content from the customer:

```jsx
if (features.ENROLL_WITH_CODE && offerCatalogs.length > 0) {
  // shows catalogs for which a user has 100% vouchers
  return getCatalogString(offerCatalogs);
}

return `enterprise_customer_uuids:${enterpriseConfig.uuid}`;
```

If our default filter relies on offers when a user does not have any assigned, we run the risk of not supplying a filter at all to Algolia, possibly showing all content across all customers.